azext_iot/dps/providers/discovery.py

```diff
@@ -0,0 +1,73 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azext_iot.common._azure import IOT_SERVICE_CS_TEMPLATE
+from azext_iot.common.base_discovery import BaseDiscovery
+from azext_iot.common.shared import DiscoveryResourceType
+from azext_iot._factory import iot_service_provisioning_factory
+from typing import Any, Dict
+
+logger = get_logger(__name__)
+PRIVILEDGED_ACCESS_RIGHTS_SET = set(
+    ["ServiceConfig", "EnrollmentWrite"]
+)
+
+
+class DPSDiscovery(BaseDiscovery):
+    def __init__(self, cmd):
+        super().__init__(
+            cmd=cmd,
+            necessary_rights_set=PRIVILEDGED_ACCESS_RIGHTS_SET,
+            resource_type=DiscoveryResourceType.DPS.value
+        )
+
+    def _initialize_client(self):
```

Comment:
We need to ensure the Track 2 DPS mgmt SDK is also supported  CC: @c-ryan-k

Reply:
Here is the link to the 1.0.0 sdk: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/iothub/azure-mgmt-iothubprovisioningservices/azure/mgmt/iothubprovisioningservices

Regarding line 34, it seems like iot_dps_resource is still used https://github.com/Azure/azure-sdk-for-python/blob/0a0cc97f178a7476ec79f29c090b8c93ad5d4955/sdk/iothub/azure-mgmt-iothubprovisioningservices/azure/mgmt/iothubprovisioningservices/_iot_dps_client.py#L67

Regarding line 42, the get_keys_for_key_name is still called list_keys_for_key_name https://github.com/Azure/azure-sdk-for-python/blob/0a0cc97f178a7476ec79f29c090b8c93ad5d4955/sdk/iothub/azure-mgmt-iothubprovisioningservices/azure/mgmt/iothubprovisioningservices/operations/_iot_dps_resource_operations.py#L925

There is a track2 flag for the poller operations. We can check the sdk version the same way we do for iot hub.

Reply:
Apply verification tests and changes if needed in next iteration.